### PR TITLE
feat: receive isThemeMode from storm

### PR DESCRIPTION
**Pull request by asanehisa**

### CODE REVIEW

File: src/internal/components/InternalEditor.tsx [hunk 1]
```diff
@@ -48,7 +48,9 @@ export const InternalEditor = ({
   themeConfig,
 }: InternalEditorProps) => {
   const [canEdit, setCanEdit] = useState<boolean>(false);
-  const [themeModeActive, setThemeModeActive] = useState<boolean>(false);
+  const [themeModeActive, setThemeModeActive] = useState<boolean>(
+    !!templateMetadata.isThemeMode
```
Comment (mkilpatrick) [inline src/internal/components/InternalEditor.tsx:52]: Is the coalesce necessary?

File: src/internal/puck/components/Header.tsx [hunk 2]
```diff
@@ -55,10 +56,12 @@ export const customHeader = (
       <div className="header-left">
         <ToggleUIButtons />
         <ToggleEntityFields />
-        <ToggleThemeMode
-          themeModeActive={themeModeActive}
-          toggleThemeMode={setThemeModeActive}
-        />
+        {!isThemeMode && (
+          <ToggleThemeMode
```
Comment (mkilpatrick) [inline src/internal/puck/components/Header.tsx:60]: This is going to be removed later?

File: src/internal/components/InternalEditor.tsx (hunk 1 above)
Comment (asanehisa) [reply src/internal/components/InternalEditor.tsx:52]: the !! is b/c I didn't wanna make isThemeMode required (for now), so it doesn't break sites w/o changes.

File: src/internal/puck/components/Header.tsx (hunk 2 above)
Comment (asanehisa) [reply src/internal/puck/components/Header.tsx:60]: Oh we don't want the toggleThemeMode toggle ever? If so, then yes I can remove it later

Comment (mkilpatrick) [reply src/internal/puck/components/Header.tsx:60]: Nope, it will be set from getting to it via the nav instead

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**